### DISCUSSION TITLE
Typescript & componentsjs-generator setup help

### DISCUSSION THREAD

**Discussion by ixuz** [LinkedSoftwareDependencies/Components.js#56] [Q&A] [answered]
Hi there!

I'm having some problems setting up a typescript project that uses `typescript`, `componentsjs` & `componentsjs-generator`.
I've attempted to follow the [componentsjs docs](https://componentsjs.readthedocs.io/en/latest/getting_started/basics/workflow_ts/) and the [componentsjs-generator docs](https://github.com/LinkedSoftwareDependencies/Components-Generator.js/blob/master/README.md) but it seems like I'm getting some configuration/detail(s) wrong.

My project: https://github.com/ixuz/componentsjs-ts-start

I'm running into this error when starting `npm run prepare && npm start`:
```
ErrorResourcesContext: Could not find (valid) component types for config "urn:componentsjs-ts-start:default:App" among its types, or a requireName
```

Additionally, see: 
[componentsjs-error-state.json.txt](https://github.com/LinkedSoftwareDependencies/Components.js/files/6953501/componentsjs-error-state.json.txt)

Any advice/help would be much appreciated!
Thanks! 😄

**Comment by rubensworks** [comment, marked as answer]
Your overall setup looks correct to me, except for one thing:

I suspect the problem lies here:
```ts
    const componentsManager = await ComponentsManager.build({
      mainModulePath: __dirname, // Path to your npm package's root
      moduleLoader: (registry) =>
        registry.registerModule("config/config.jsonld"),
    });
```

You are overriding the default `moduleLoader` (the thing that loads the instantiatable components into a registry), and are registering your config as a module (while this isn't a module, but a config).
You can probably just keep the default `moduleLoader` (not override it), as this will load all possible modules it can find, so that you can refer to them from your config file.

In order to register your config, the commented-out line is in fact what you'll need:
```ts
await componentsManager.configRegistry.register("config/config.jsonld");
```

**Reply by ixuz** [reply]
Wow, super quick answer and thank you!
Yes, that makes sense, I've applied your fixes/recommendation(https://github.com/ixuz/componentsjs-ts-start/commit/28cd462f696f22a5271336f0c8e294b20b6dafeb), but there seems to be another issue.

This is the error that came up:

```
node:internal/modules/cjs/loader:933
  const err = new Error(message);
              ^

Error: Cannot find module 'componentsjs-ts-start'
Require stack:
- E:\Dev\componentsjs-ts-start\node_modules\componentsjs\lib\loading\ComponentsManagerBuilder.js
- E:\Dev\componentsjs-ts-start\node_modules\componentsjs\index.js
- E:\Dev\componentsjs-ts-start\dist\AppRunner.js
- E:\Dev\componentsjs-ts-start\dist\index.js
- E:\Dev\componentsjs-ts-start\bin\server.js
```

**Reply by rubensworks** [reply]
I suspect this is because your `mainModulePath` is not pointing to the root of your package.
Something like this should do the trick: `mainModulePath: Path.join(__dirname, '../..')`

**Reply by ixuz** [reply]
I managed to instantiate the `App` now by adjusting the `mainModulePath`, see: https://github.com/ixuz/componentsjs-ts-start/commit/217dff2bcf059ab352021c66d0593354ffd65c49

Thank you for the help @rubensworks ! 😃

**Reply by rubensworks** [reply]
Great, you're welcome!
